cvat/apps/engine/serializers.py

```diff
@@ -1207,6 +1206,65 @@ class SubLabeledShapeSerializer(ShapeSerializer, AnnotationSerializer):
 class LabeledShapeSerializer(SubLabeledShapeSerializer):
     elements = SubLabeledShapeSerializer(many=True, required=False)
 
+class LabeledImageSerializerFromDB(serializers.BaseSerializer):
+    def to_representation(self, instance):
+        def convert_tag(tag):
+            tag['attributes'] = [OrderedDict(attr) for attr in tag['labeledimageattributeval_set']]
+            del tag['labeledimageattributeval_set']
+            if 'parent' in tag:
+                del tag['parent']
+            for attr in tag['attributes']:
+                del attr['id']
+
+            return OrderedDict(tag)
```

Comment:
This function seems easy to misuse, because it both modifies the old value _and_ returns a new one. Normally, I would expect it to do only one of those.

Why not have the function construct a new dictionary with the needed keys and return it? It would be easier to understand, and also safer, because you won't be able to expose unwanted fields by forgetting to delete them.

Reply:
Makes sense